flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -404,10 +437,15 @@ void createIcebergTable(ObjectPath tablePath, ResolvedCatalogTable table, boolea
     ImmutableMap.Builder<String, String> properties = ImmutableMap.builder();
     String location = null;
     for (Map.Entry<String, String> entry : table.getOptions().entrySet()) {
-      if ("location".equalsIgnoreCase(entry.getKey())) {
-        location = entry.getValue();
-      } else {
+      if (!isReservedProperty(entry.getKey())) {
         properties.put(entry.getKey(), entry.getValue());
+      } else {
+        // Filtering reserved properties like catalog properties(added to support CREATE TABLE LIKE
+        // in getTable()),
+        // location and not persisting on table properties.
+        if ("location".equalsIgnoreCase(entry.getKey())) {
```

Comment:
move "location" literal to a constant like others

Reply:
location is being used at too many places across iceberg code. Not sure what would be right place and do we want to change across all usages ?

Reply:
I was thinking about just this class in Flink module. we don't want to change all usages.